apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java

```diff
@@ -148,19 +148,11 @@ public static void setUsernameIfUnset(@Nullable String userName, TransactionCont
     public void onAfter(Transaction transaction, @Nullable Throwable exception, boolean committed, int status, String method,
                         @Nullable Map<String, String[]> parameterMap, String servletPath, @Nullable String pathInfo, @Nullable String contentTypeHeader) {
         try {
-            fillRequestParameters(transaction, method, parameterMap, contentTypeHeader);
-            if(exception != null && status == 200) {
-                // Probably shouldn't be 200 but 5XX, but we are going to miss this...
-                status = 500;
-            }
-            fillResponse(transaction.getContext().getResponse(), committed, status);
-            transaction.withResult(ResultUtil.getResultByHttpStatus(status));
-            transaction.withType("request");
-            if (transaction.getName().length() == 0) {
-                applyDefaultTransactionName(method, servletPath, pathInfo, transaction.getName());
-            }
-            if (exception != null) {
-                transaction.captureException(exception);
+            // thrown the first time a JSP is invoked in order to register it
+            if (exception != null && "weblogic.servlet.jsp.AddToMapException".equals(exception.getClass().getName())) {
```

Comment:
We agreed not to add platform/library/framework-specific code into general agent code

Reply:
The case we discussed was to not add JSF specific stuff in the servlet plugin. But this is still Servlet, though also WebLogic specific. We already have application server specifics, for example in

https://github.com/elastic/apm-agent-java/blob/39ee1f05e213c090d7c27b18b30b39d3bc09998d/apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/ServiceNameUtil.java#L58-L71

and

https://github.com/elastic/apm-agent-java/blob/39ee1f05e213c090d7c27b18b30b39d3bc09998d/apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/helper/ApmAsyncListener.java#L119-L122

So I think servlet container/application server specifics are ok to have directly in the servlet plugin.

Reply:
I think they are all similar.
If this makes everything much simpler then I don't have a problem with it, but if it's easy to separate then it is probably better.
Maybe when it gets too loaded with such stuff we can add an app-server-specific API for which a factory chooses an implementation based on the app-server on startup.
I don't think you should waste efforts on this, only move if very simple.